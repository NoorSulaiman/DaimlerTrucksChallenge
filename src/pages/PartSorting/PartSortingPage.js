import React, { useState, useContext } from "react";
import { appContext } from "../../provider";
import GroupsList from "../../components/GroupsList/GroupsList";
import ComponentsDetailsList from "../../components/ComponentsList/ComponentsDetailsList";

export default function PartSortingPage() {
  const context = useContext(appContext);
  const groupsArray = context.parts;
  const [componentsList, setComponentsList] = useState([]);

  const sortComponents = (group) => {
    let sortedComponentsArray = [];
    groupsArray.map((component) => {
      let prefix = component.location.slice(0, 2);
      if (group === prefix) {
        sortedComponentsArray.push(component);
      }
    });

    sortedComponentsArray = sortedComponentsArray.sort((a, b) =>
      a.location > b.location ? 1 : b.location > a.location ? -1 : 0
    );
    setComponentsList(sortedComponentsArray);
  };
  return (
    <div>
      {groupsArray.length ? (
        <GroupsList components={groupsArray} selectGroup={sortComponents} />
      ) : (
        "No Groups Found..."
      )}
      {componentsList.length ? (
        <ComponentsDetailsList components={componentsList} />
      ) : null}
    </div>
  );
}
