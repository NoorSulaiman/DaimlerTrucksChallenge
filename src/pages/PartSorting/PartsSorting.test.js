import React from "react";
import { shallow } from "enzyme";
import PartSortingPage from "./PartSortingPage";

const compArray = [
  {
    panelNumber: "22-58883-000",
    panelDescription: "PNL-INR TRIMPLTKAHLUA SPRKL",
    location: "AGAAA",
    parentPanelNumber: "734-C02202",
    modelCode: "H",
    whitelist: ["AGAAA"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "",
    bom: "734-C02202",
    isAsBuilt: true,
  },
  {
    panelNumber: "22-69410-000",
    panelDescription: "PNL-SW,SMARTPLX,OVHD,SD,KAHLUA",
    location: "OVLHP",
    parentPanelNumber: "679-C01232",
    modelCode: "H",
    whitelist: ["OVLHP", "OVRHP"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "",
    bom: "679-C01232",
    isAsBuilt: true,
  },
  {
    panelNumber: "A06-30769-027",
    panelDescription: "SW-RCKR,PS PWR WDW",
    location: "AG017",
    parentPanelNumber: "66A-C00132",
    modelCode: "H",
    whitelist: [
      "AG003",
      "AG004",
      "AG009",
      "AG010",
      "AG013",
      "AG014",
      "AG015",
      "AG016",
      "AG017",
      "AG018",
      "AG019",
      "AG023",
      "AG024",
      "AG025",
      "AG026",
      "AG027",
      "AG028",
      "AG096",
      "AG097",
    ],
    blacklist: [],
    type: "hardwired-component",
    switchId: "NONE",
    bom: "66A-C00132",
    isAsBuilt: true,
  },
];
const sortComponents = (group) => {
  let sortedComponentsArray = [];
  compArray.map((component) => {
    let prefix = component.location.slice(0, 2);
    if (group === prefix) {
      sortedComponentsArray.push(component);
    }
  });

  sortedComponentsArray = sortedComponentsArray.sort((a, b) =>
    a.location > b.location ? 1 : b.location > a.location ? -1 : 0
  );
  return sortedComponentsArray;
};
describe("PartSortingPage", () => {
  it("should render PartSorting Page correctly with given props", () => {
    const component = shallow(<PartSortingPage />);
    expect(component).toMatchSnapshot();
  });

  it("should sort components list into group prefixes", () => {
    const component = shallow(<PartSortingPage />);
    expect(component.find("GroupsList").length).toBe(1);
  });

  it("should take prefixs and sort them", () => {
    expect(sortComponents("AG").length).toStrictEqual(2);
  });
});
