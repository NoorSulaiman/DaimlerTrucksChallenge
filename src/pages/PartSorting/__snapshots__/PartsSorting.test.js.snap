// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartSortingPage should render PartSorting Page correctly with given props 1`] = `ShallowWrapper {}`;
